hawkular-agent-core/src/main/java/org/hawkular/agent/monitor/storage/HawkularStorageAdapter.java

```diff
@@ -72,11 +73,13 @@ public void initialize(
             AgentCoreEngineConfiguration.StorageAdapterConfiguration config,
             int autoDiscoveryScanPeriodSeconds,
             Diagnostics diag,
-            HttpClientBuilder httpClientBuilder) {
+            HttpClientBuilder httpClientBuilder,
+            BaseRestServerGenerator restServerGenerator) {
```

Comment:
@jmazzitelli 
Are you ok with this change?

I need to build the BaseRestServerGenerator in the AgentCoreEngine to pass the SSLContext.